mtags/src/main/scala-3/scala/meta/internal/pc/completions/MatchCaseCompletions.scala

```diff
@@ -86,34 +86,42 @@ object CaseKeywordCompletion:
             new Parents(argTypes, definitions)
           case _ =>
             new Parents(NoType, definitions)
-      case sel => new Parents(sel.tpe, definitions)
+      case sel =>
+        val selectorTpe = sel.tpe
+        new Parents(selectorTpe, definitions)
 
     val selectorSym = parents.selector.typeSymbol
-
     // Special handle case when selector is a tuple or `FunctionN`.
     if definitions.isTupleClass(selectorSym) || definitions.isFunctionClass(
         selectorSym
       )
     then
+      val selectorTpe = parents.selector.show
+      val tpeLabel =
+        if !selectorTpe.contains("x$1") then selectorTpe
+        else selector.symbol.info.show
       val label =
-        if patternOnly.isEmpty then s"case ${parents.selector.show} =>"
-        else parents.selector.show
-      List(
-        CompletionValue.CaseKeyword(
-          selectorSym,
-          label,
-          Some(
-            if patternOnly.isEmpty then
-              if config.isCompletionSnippetsEnabled() then "case ($0) =>"
-              else "case () =>"
-            else if config.isCompletionSnippetsEnabled() then "($0)"
-            else "()"
-          ),
-          Nil,
-          range = Some(completionPos.toEditRange),
-          command = config.parameterHintsCommand().asScala,
+        if patternOnly.isEmpty then s"case ${tpeLabel} =>"
+        else tpeLabel
+      if completionGenerator.fuzzyMatches(label) then
```

Comment:
Maybe `patternOnly.isEmpty || patternOnly.contains(CURSOR)`?

`val a = List(("a", 1), ("b", 2), ("c", 2)).map{ case (Strin@@) =>}`
Probably shouldn't match, since it will put just `()`.

Reply:
Actually I'm now thinking if we should provide tuple completions in for pattern only at all. Currently I don't see any way to distinguish between `case (@@) =>` and `case @@ =>`, and we don't to give this completion in the first case. Only problem I see is that this completion may suggest to the user what is the type of the argument, so it can be useful

Reply:
Maybe it makes not sense for pattern only in that case. I don't think that suggesting types is a responsibility of a completion.